### PR TITLE
Refactor InitCodeGenerator to provide a single entry point.

### PR DESCRIPTION
**Pull request by shinfan**
- InitCodeGenerator now provides a single entry point which reduces duplicate code and improves readability.
- The init code generation is fully configured by InitCodeGeneratorContext.

### CODE REVIEW

File: src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java [hunk 1]
```diff
@@ -84,9 +89,11 @@ public InitCodeView generateRequestObjectInitCode(
             .valueGenerator(valueGenerator)
             .initStructure(initFieldStructure)
             .method(context.getMethod())
+            .initObjectType(context.getMethod().getInputType())
+            .initObjectName(Name.from("request"))
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java]: Be consistent with ordering - everywhere else has 'initObjectName' before 'initObjectType'

File: src/main/java/com/google/api/codegen/metacode/InitCodeGenerator.java [hunk 2]
```diff
@@ -35,77 +33,42 @@
...
-    FieldSetting requestField =
-        FieldSetting.create(
-            method.getInputType(),
-            Name.from("request"),
-            lastLine.getIdentifier(),
-            lastLine.getInitValueConfig());
-    List<FieldSetting> outputFields = Arrays.asList(requestField);
-    return InitCode.create(initLineSpecs, outputFields);
-  }
+  public InitCode generate(InitCodeGeneratorContext context) {
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/metacode/InitCodeGenerator.java]: Add a doc

**Comment by garrettjonesgoogle** [comment]
Nice refactoring! Just a couple nits.

File: src/main/java/com/google/api/codegen/metacode/InitCodeGenerator.java (hunk 2 above)
Comment (shinfan) [inline src/main/java/com/google/api/codegen/metacode/InitCodeGenerator.java]: Done.

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**